src/Compatibility/Microsoft.DotNet.ApiCompatibility/Abstractions/CompatDifference.cs

```diff
@@ -9,7 +9,7 @@ namespace Microsoft.DotNet.ApiCompatibility.Abstractions
     /// <summary>
     /// Class representing a difference of compatibility, containing detailed information about it.
     /// </summary>
-    public class CompatDifference : IDiagnostic, IEquatable<CompatDifference>
```

Comment:
What was the reasoning behind removing the IEquatable here? I see you are now moving it to the tests but wouldn't it still make sense to keep it in here in case there is ever a need to diff to CompatDifferences? In particular, I think that this may be useful when dealing with suppressions.